src/main/java/org/opensearch/neuralsearch/common/MinClusterVersionUtil.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.common;
+
+import com.google.common.collect.ImmutableMap;
+import org.opensearch.Version;
+import org.opensearch.knn.index.IndexUtil;
+import org.opensearch.neuralsearch.util.NeuralSearchClusterUtil;
+
+import java.util.Map;
+
+import static org.opensearch.knn.index.query.KNNQueryBuilder.MAX_DISTANCE_FIELD;
+import static org.opensearch.knn.index.query.KNNQueryBuilder.MIN_SCORE_FIELD;
+import static org.opensearch.neuralsearch.query.NeuralQueryBuilder.MODEL_ID_FIELD;
+
+public class MinClusterVersionUtil {
+
+    private static final Version MINIMAL_SUPPORTED_VERSION_DEFAULT_MODEL_ID = Version.V_2_11_0;
+    private static final Version MINIMAL_SUPPORTED_VERSION_RADIAL_SEARCH = Version.V_2_14_0;
+
+    // Note this minimal version will act as a override
+    private static final Map<String, Version> MINIMAL_VERSION_NEURAL = ImmutableMap.<String, Version>builder()
+        .put(MODEL_ID_FIELD.getPreferredName(), MINIMAL_SUPPORTED_VERSION_DEFAULT_MODEL_ID)
+        .put(MAX_DISTANCE_FIELD.getPreferredName(), MINIMAL_SUPPORTED_VERSION_RADIAL_SEARCH)
+        .put(MIN_SCORE_FIELD.getPreferredName(), MINIMAL_SUPPORTED_VERSION_RADIAL_SEARCH)
+        .build();
+
+    public static boolean isClusterOnOrAfterMinReqVersionForDefaultModelIdSupport() {
+        return NeuralSearchClusterUtil.instance().getClusterMinVersion().onOrAfter(MINIMAL_SUPPORTED_VERSION_DEFAULT_MODEL_ID);
+    }
+
+    public static boolean isClusterOnOrAfterMinReqVersionForRadialSearch() {
+        return NeuralSearchClusterUtil.instance().getClusterMinVersion().onOrAfter(MINIMAL_SUPPORTED_VERSION_RADIAL_SEARCH);
+    }
+
+    public static boolean isClusterOnOrAfterMinReqVersion(String key) {
+        Version version;
+        if (MINIMAL_VERSION_NEURAL.containsKey(key)) {
+            version = MINIMAL_VERSION_NEURAL.get(key);
+        } else {
+            version = IndexUtil.minimalRequiredVersionMap.get(key);
```

Comment:
IndexUtil is from knn repo, we shouldn't add/keep to minimal such dependencies on other plugins.

Also for the logic itself - why we're falling back to knn implementation in case we cannot find settings specific to neural search? I think we either add settings to neural or return 'false'

Reply:
> IndexUtil is from knn repo, we shouldn't add/keep to minimal such dependencies on other plugins.

Isn't this a wrapper on k-NN and takes a dependency on it? Why can't there be class dependencies for reuse when its a direct dependency?

> Also for the logic itself - why we're falling back to knn implementation in case we cannot find settings specific to neural search? I think we either add settings to neural or return 'false'

To not have a mismatch for Knn related parameters. For instance, efsearch has a version associated with it. Inconsistencies might cause issues there. If we want it to be different we should add it in `MINIMAL_VERSION_NEURAL`.